command/head.go

```diff
@@ -0,0 +1,313 @@
+	showFullPath     bool
+
+	storageOpts storage.Options
+}
+
+func (h Head) Run(ctx context.Context) error {
+	h.src.SetRelative(h.src)
+	client, err := storage.NewRemoteClient(ctx, h.src, h.storageOpts)
+	if err != nil {
+		printError(h.fullCommand, h.op, err)
+		return err
+	}
+
+	if h.src.IsBucket() {
+		err := client.HeadBucket(ctx, h.src)
+		if err != nil {
+			printError(h.fullCommand, h.op, err)
+			return err
+		}
+
+		msg := HeadBucketMessage{
+			Name: h.src.String(),
+		}
+
+		log.Info(msg)
+
+		return nil
+	}
+
+	object, metadata, err := client.HeadObject(ctx, h.src)
+	if err != nil {
+		printError(h.fullCommand, h.op, err)
+		return err
+	}
+
+	msg := HeadObjectMessage{
+		Object:           object,
+		showEtag:         h.showEtag,
+		showHumanized:    h.humanize,
+		showStorageClass: h.showStorageClass,
+		showFullPath:     h.showFullPath,
+		Metadata:         metadata,
+	}
+
+	log.Info(msg)
+
+	return nil
+}
+
+type HeadObjectMessage struct {
+	Object *storage.Object `json:"object"`
+
+	showEtag         bool
+	showHumanized    bool
+	showStorageClass bool
+	showFullPath     bool
+	Metadata         map[string]string `json:"metadata"`
+}
+
+func (m HeadObjectMessage) String() string {
+	if m.showFullPath {
+		return m.Object.URL.String()
+	}
+	var etag string
+	// date and storage fields
+	listFormat := "%19s %2s"
+
+	// align etag
+	if m.showEtag {
+		etag = m.Object.Etag
+		listFormat = listFormat + " %-38s"
+	} else {
+		listFormat = listFormat + " %-1s"
+	}
+
+	// format file size
+	listFormat = listFormat + " %12s "
+	// format key and version ID
+	if m.Object.URL.VersionID != "" {
+		listFormat = listFormat + " %-50s %s"
+	} else {
+		listFormat = listFormat + " %s%s"
+	}
+
+	var s string
+	if m.Object.Type.IsDir() {
+		s = fmt.Sprintf(
+			listFormat,
+			"",
+			"",
+			"",
+			"DIR",
+			m.Object.URL.Relative(),
+			"",
+		)
+		return s
+	}
+
+	stclass := ""
+	if m.showStorageClass {
+		stclass = fmt.Sprintf("%v", m.Object.StorageClass)
+	}
+
+	var path string
+	if m.showFullPath {
+		path = m.Object.URL.String()
+	} else {
+		path = m.Object.URL.Relative()
+	}
+
+	s = fmt.Sprintf(
+		listFormat,
+		m.Object.ModTime.Format(dateFormat),
+		stclass,
+		etag,
+		m.humanize(),
+		path,
+		m.Object.URL.VersionID,
+	)
+
+	return s
+}
+
+func (m HeadObjectMessage) JSON() string {
+	j := struct {
+		storage.Object
+		Metadata map[string]string `json:"metadata"`
+	}{
+		Object:   *m.Object,
+		Metadata: m.Metadata,
+	}
+
+	return strutil.JSON(j)
+}
+
+func (m HeadObjectMessage) humanize() string {
+	var size string
+	if m.showHumanized {
+		size = strutil.HumanizeBytes(m.Object.Size)
+	} else {
+		size = fmt.Sprintf("%d", m.Object.Size)
+	}
+	return size
+}
+
+type HeadBucketMessage struct {
+	Name string `json:"name"`
+}
+
+func (m HeadBucketMessage) String() string {
+	return fmt.Sprintf(m.Name)
+}
+
+func (m HeadBucketMessage) JSON() string {
+	return strutil.JSON(m)
+}
+
+func validateHeadCommand(c *cli.Context) error {
```

Comment:
There're a few scenarios where error messages are not very helpful.

1. Running the command on a local file:

```bash
$/ ./s5cmd head go.mod
ERROR "head go.mod": InvalidParameter: 1 validation error(s) found. - minimum field size of 1, HeadObjectInput.Bucket.
```

2. Running the command on an S3 prefix:

```bash
$ ./s5cmd head s3://bigtmp/ibrahim/
ERROR "head s3://bigtmp/ibrahim/": given object s3://bigtmp/ibrahim/ not found
```

Expected an error like "target must a bucket or an object"

3. Running the command without an argument:

```bash
./s5cmd head
ERROR "head": InvalidParameter: 2 validation error(s) found. - minimum field size of 1, HeadObjectInput.Bucket. - minimum field size of 1, HeadObjectInput.Key.
```

Expected something like "expected a bucket or an object"